frontend/public/components/monitoring/alert-manager-utils.tsx

```diff
@@ -36,6 +43,179 @@ export const patchAlertManagerConfig = (
 };
 
 export const receiverTypes = {
-  pagerduty: 'PagerDuty',
-  webhook: 'Webhook Receiver',
+  pagerduty_configs: 'PagerDuty',
+  webhook_configs: 'Webhook Receiver',
+};
+
+export const isKnownReceiverType = (receiverType: string) => {
+  return _.hasIn(receiverTypes, `${receiverType}_configs`);
+};
+
+/**
+ * Converts route object:
+ * {
+ *   receiver: "MyReceiver",
+ *   match: {
+ *     severity: "warning",
+ *     cluster: "myCluster"
+ *   },
+ *   match_re: {
+ *    service: "$foobar"
+ *  }
+};
+ * ...to array of labels for Routing Labels form component
+ * [
+ *   {
+ *     "name": "severity",
+ *     "value": "warning",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "cluster",
+ *     "value": "myCluster",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "service",
+ *     "value": "$foobar",
+ *     "isRegex": true
+ *   }
+ * ]
+ */
+export const getRouteLabelsForForm = (
+  receiver: AlertManagerReceiver,
+  routes: AlertManagerRoute[],
+) => {
+  const routesOfReceiver = _.find(
+    routes,
+    (aRoute: AlertManagerRoute) => aRoute.receiver === receiver.name,
+  );
+  const matches = _.map(_.get(routesOfReceiver, 'match', {}), (v, k) => {
+    return { name: k, value: v, isRegex: false };
+  });
+  const regexMatches = _.map(_.get(routesOfReceiver, 'match_re', {}), (v, k) => {
+    return { name: k, value: v, isRegex: true };
+  });
+  return _.concat([], matches, regexMatches);
+};
+
+/**
+ * Converts array of labels (from Form):
+ * [
+ *   {
+ *     "name": "severity",
+ *     "value": "warning",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "cluster",
+ *     "value": "myCluster",
+ *     "isRegex": false
+ *   }
+ * ]
+ * ...to single label object, for Patch:
+ * {
+ *   "severity": "warning",
+ *   "cluster": "myCluster"
+ * }
+ */
+export const getRouteLabelsForPatch = (routeLabels, returnRegexMatches = false) => {
```

Comment:
Looks like `getRouteLabelsForPatch` is now only used by `createRoute`, so no need to export (or maybe refactor it away as suggested below).

Reply:
removed `getRouteLabelsForPatch`